packages/color-mode/src/color-mode.utils.ts

```diff
@@ -75,12 +75,13 @@ export function addListener(
 
 export const root = {
   get: () =>
-    document.documentElement.style.getPropertyValue(
+    (document.documentElement.style.getPropertyValue(
       "--chakra-ui-color-mode",
-    ) as ColorMode | "",
+    ) || document.documentElement.dataset.theme) as ColorMode | "",
   set: (mode: ColorMode) => {
     if (isBrowser) {
       document.documentElement.style.setProperty("--chakra-ui-color-mode", mode)
+      document.documentElement.setAttribute("data-theme", mode)
```

Comment:
I am syncing the current color mode to the document root: e.g. `<html data-theme="dark">`.
This allows the `_dark` pseudo selector (`".chakra-ui-dark &, [data-theme=dark] &, &[data-theme=dark]"`) to pick up the current color mode.

Reply:
As discussed with Sage we are going to deprecate the internal CSS var `--chakra-ui-color-mode` with the next major release.